service/src/main/java/com/ge/predix/acs/privilege/management/dao/GraphGenericRepository.java

```diff
@@ -402,7 +401,7 @@ private void searchAttributesWithScopes(final E entity, final Vertex vertex, fin
                         attributes.addAll(deserializedAttributes);
                         // This enforces the limit on the count of attributes returned from the traversal, instead of
                         // number of vertices traversed. To do the latter will require traversing the graph twice.
-                        checkTraversalLimitOrFail(attributes);
+                        checkTraversalLimitOrFail(entity, attributes);
```

Comment:
to be clear, is this what you are proposing ?

* remove limit() from graph traversal
* replace line 404 with:

>  int traversalCount;
>  if (++traversalCount > this .traversalLimit)  {
>     throw new AttributeTraversalLimitExceededException()...
>  }